docs/DeveloperGuide.md

```diff
@@ -1,29 +1,159 @@
 # Developer Guide
+The aim of this guide is to help readers understand how the system and components of RecipEditor 
+is designed, implemented and tested. This developer guide also serves to help 
+developers to understand the architecture of RecipEditor and some design considerations. 
+Click to view the latest release of [RecipEditor]((https://github.com/AY2223S1-CS2113-T18-2/tp/releases)).
+
+## Content page
+[Acknowledgements](#acknowledgements)
+
+[Design](#design)
+- [Architecture](#architecture)
+- [Logic Component](#logic-component)
+- [Ui Component](#ui-component)
+- [Storage Component](#storage-component)
+- [Implementation](#implementation)
+    - [Loading Of Data On Startup](#loading-of-data-on-startup)
+    - [Parsing of Commands](#parsing-of-commands)
+    - [Add Recipe](#add-recipe)
+    - [Edit an Existing Recipe](#add-an-existing-recipe)
+    - [Find Recipe Based on Recipe Name and Ingredient](#find-recipe-based-on-recipe-name-and-ingredient)
+- [Product Scope](#product-scope)
+    - [Target User Profile](#target-user-profile)
+    - [Value Proposition](#value-proposition)
+- [User Stories](#user-stories)
 
 ## Acknowledgements
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+### External Libraries
+- org.apache.commons:commons-lang3:3.0 [link](https://mvnrepository.com/artifact/org.apache.commons/commons-lang3/3.0)
+- org.apiguardian:apiguardian-api:1.1.0 [link](https://mvnrepository.com/artifact/org.apiguardian/apiguardian-api)
 
-## Design & implementation
+## Design
+### Architecture
```

Comment:
Maybe you can give a more high level overview of class component diagram in the Architecture. Right now the sequence diagram is more of implementation